### PR TITLE
Session persistent support

### LINKED ISSUE #7: Persist session + connection timeout
we kill the session immediately after the app enters background, a better way is launching a background task and alert user to go back to the app, and only kill the sessions when user press disconnect button explicitly.

for the white screen issue, we can add a timeout if server doesn't send a session request and kill the session

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #7

### CODE REVIEW

**Comment by leoneparise** [comment]
> Fixes #7

Can we add some unit tests?

File: Example/WalletConnect/WCSessionViewController.swift [hunk 1]
```diff
@@ -129,7 +152,7 @@ class ViewController: UIViewController {
 
         var result = privateKey.sign(digest: Hash.keccak256(data: data), curve: .secp256k1)!
         result[64] += 27
-        self.interactor?.approveRequest(id: id, result: result.hexString).cauterize()
+        self.interactor?.approveRequest(id: id, result: "0x" + result.hexString).cauterize()
```
Comment (vikmeup) [inline Example/WalletConnect/WCSessionViewController.swift]: did you consider adding 0x prefix extension?

Comment (vikmeup) [reply Example/WalletConnect/WCSessionViewController.swift]: `result.hexString.0xPrefix` or something similar?  `result.0xhex`

File: Example/WalletConnect/WCSessionViewController.swift [hunk 2]
```diff
@@ -182,3 +213,71 @@ class ViewController: UIViewController {
...
+
+    func startBackgroundTask(_ application: UIApplication) {
+        backgroundTaskId = application.beginBackgroundTask(withName: "WalletConnect", expirationHandler: {
+            self.backgroundTimer?.invalidate()
+            print("<== background task expired")
+        })
+
+        var alerted = false
+        backgroundTimer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
+            print("<== background time remainning: ", application.backgroundTimeRemaining)
```
Comment (vikmeup) [inline Example/WalletConnect/WCSessionViewController.swift]: does this trigger background timer to fire many times when you come up? we had that issue before

**Comment by hewigovens** [comment]
@leoneparise sure, let me add codecov too

File: Example/WalletConnect/WCSessionViewController.swift (hunk 1 above)
Comment (hewigovens) [reply Example/WalletConnect/WCSessionViewController.swift]: we can add that, but it's an example app, worth to do it?

File: Example/WalletConnect/WCSessionViewController.swift (hunk 2 above)
Comment (hewigovens) [reply Example/WalletConnect/WCSessionViewController.swift]: we have code in `applicationWillEnterForeground`

```swift
if let id = backgroundTaskId {
    application.endBackgroundTask(id)
}
backgroundTimer?.invalidate()
```

should be enough?

**Pull request merged**